src/wallet/wallet2.cpp

```diff
@@ -1206,8 +1206,22 @@ void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& re
   get_short_chain_history(short_chain_history);
   m_run.store(true, std::memory_order_relaxed);
   if (start_height > m_blockchain.size() || m_refresh_from_block_height > m_blockchain.size()) {
-    if (!start_height)
+    if (!start_height) {
+
+      std::string daemon_height_err = "";
+      uint64_t daemon_bc_height = this->get_daemon_blockchain_height(daemon_height_err);
```

Comment:
What happens when the daemon isn't fully synced? It looks like it would set m_refresh_from_block_height to current daemon height, abort fast_refresh() on that block and continue refresh with the slower pull_blocks(). Not sure why we would wan't that? 
Maybe wallet2::get_daemon_blockchain_target_height() could be used instead?